frontend/apps/filemanager/filemanager.lua

```diff
@@ -377,6 +377,8 @@ function FileManager:registerKeyEvents()
         self.file_chooser.key_events.Back = { { Device.input.group.Back } }
         if Device:hasScreenKB() then
             self.key_events.ToggleWifi = { { "ScreenKB", "Home" } }
+        elseif Device:hasKeyboard() then
+            self.key_events.ShowFileSearch = { { "Shift", "S" }, event = "ShowFileSearch" }
```

Comment:
We (you) usually register key events in the modules that contain the event handler.
Doing so would allow you to call file searcher from the Reader (consistently, we can call it via a gesture in FM and in the Reader).

But... Is there any negative output of transforming many WidgetContainers to InputContainers (memory consumption etc)?
If yes, maybe it is worth to accumulate key_events in the parent modules (FM, Reader), even duplicating?

In any case, we need a consistent approach, not a chaos.

Reply:
But... Is there any negative output of transforming many WidgetContainers to InputContainers (memory consumption etc)?

> I don't know the answer to this, what I do know is that WidgetContainers don't support `self.key_events` https://github.com/koreader/koreader/pull/11916#issuecomment-2153128144

Reply:
I don't think that should be a problem, give or take from a clarity perspective.

Reply:
> But... Is there any negative output of transforming many WidgetContainers to InputContainers (memory consumption etc)?

I don't think so (some minimal overhead when handling events as there's another object to check).

> We (you) usually register key events in the modules that contain the event handler.

I prefer this (and not: `accumulate key_events in the parent modules`).

Reply:
So it should go to filemanagerfilesearcher.